### PR TITLE
[SPARK-37622][K8S] Support K8s executor rolling policy

### PR DESCRIPTION
**Pull request by dongjoon-hyun**
### What changes were proposed in this pull request?

This PR aims to support K8s executor rolling policies via `spark.kubernetes.executor.rollPolicy`.

### Why are the changes needed?

Users can choose one of the following policy in order to decommission one executor at every `spark.kubernetes.executor.rollInterval`.
- ID: An executor with the smallest executor ID. This is most human-readable.
- ADD_TIME: An executor with the smallest add-time. This is the most long-live executor.
- TOTAL_GC_TIME: An executor with the biggest GC time. This could have GC issue for some reason.
- TOTAL_DURATION: An executor with the biggest total task time. This executor may have busy neighbor or CPU thresholding by K8s controller.

### Does this PR introduce _any_ user-facing change?

Yes, but this is a new feature.

### How was this patch tested?

Pass the CIs.

### CODE REVIEW

**Comment by dongjoon-hyun** [comment]
Could you review this please, @viirya ?

File: resource-managers/kubernetes/core/src/main/scala/org/apache/spark/deploy/k8s/Config.scala [hunk 1]
```diff
@@ -145,6 +146,20 @@ private[spark] object Config extends Logging {
...
       .createWithDefault(0)
 
+  object ExecutorRollPolicy extends Enumeration {
+    val ID, ADD_TIME, TOTAL_GC_TIME, TOTAL_DURATION = Value
+  }
+
+  val EXECUTOR_ROLL_POLICY =
+    ConfigBuilder("spark.kubernetes.executor.rollPolicy")
+      .doc("Executor roll policy: Valid values are ID, ADD_TIME, TOTAL_GC_TIME (default), " +
+        "and TOTAL_DURATION")
```
Comment (viirya) [inline resource-managers/kubernetes/core/src/main/scala/org/apache/spark/deploy/k8s/Config.scala:156]: Do we need to describe the meanings for each policy like the description does?

Comment (dongjoon-hyun) [reply resource-managers/kubernetes/core/src/main/scala/org/apache/spark/deploy/k8s/Config.scala:156]: Thank you for review. Sure! I'll add.

**Review by viirya — APPROVED** [review]
Looks a straightforward extension of executor rolling by allowing more flexibility. Just need to make sure the config is understandable well for end-user in the config doc.

**Comment by dongjoon-hyun** [comment]
I addressed your comments, @viirya . Thank you for approval.

**Comment by dongjoon-hyun** [comment]
Merged to master for Apache Spark 3.3.0.

**Pull request closed without merging**